crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2983,7 +2990,16 @@ impl<'db> TypeInferenceBuilder<'db> {
         } = import;
 
         for alias in names {
-            self.infer_definition(alias);
+            let definitions = self.index.definitions(alias);
+            if definitions.is_empty() {
+                // Ensure `from foo import *` is flagged as invalid if `foo` can't be resolved,
+                // *even if* `foo` is an empty module (and thus has no `Definition`s associated with it)
```

Comment:
This comment doesn't quite make sense to me as written. If `foo` can't be resolved, then it can't be an "empty module" -- it's a nonexistent module. It seems like more what we want to say is, unresolved modules won't have any definitions associated with them, but we still need to emit a diagnostic for their unresolvability here.